app/commands/v2/unpublish.rb

```diff
@@ -132,7 +132,9 @@ def find_unpublishable_content_item(content_id)
         end
 
         filter = ContentItemFilter.new(scope: ContentItem.where(content_id: content_id))
-        filter.filter(locale: locale, state: allowed_states).first
```

Comment:
I agree with @elliotcm ; when you publish a draft it would supersede the previous version, whether it's published or unpublished.